storage/api.go

```diff
@@ -53,16 +59,24 @@ func (b *QueryBatch) Queue(cmd string, args ...interface{}) {
 
 // Extend merges another batch into the current batch.
 func (b *QueryBatch) Extend(qb *QueryBatch) {
+	b.mu.Lock()
+	qb.mu.Lock()
+	defer b.mu.Unlock()
+	defer qb.mu.Unlock()
```

Comment:
Let's do:
```golang
	if qb != b {
		qb.mu.Lock()
		defer qb.mu.Unlock()
	}
```

That way, this won't silently deadlock in case  `Extend` is (irrationally) called on itself (e.g. `qb.Extend(&qb)`).

Reply:
Thanks! updated